lib/cocoapods/command/repo/add.rb

```diff
@@ -45,6 +45,7 @@ def run
           section << " (branch `#{@branch}`)" if @branch
           UI.section(section) do
             create_repos_dir
+            exclude_repos_dir_from_backup
```

Comment:
what would happen for existing repos?

Reply:
Since `exclude_repos_dir_from_backup` excludes the whole local spec-repos directory from Time Machine backups, all existing repos would get excluded as well.

In other words, the first time you run `pod repo add` (with these changes), `~/.cocoapods/repos` (which includes all existing repos) would get excluded from backups.

Reply:
@sonicdoe hmm I can imagine many systems already having created the repos they need. What if we put that in the `pod repo update` (`update.rb`) instead?

Reply:
In my eyes, `pod repo update` doesn’t seem like the right place since this should be a one-time operation when the repos directory is created. However, if it helps users coming from older versions and you’re fine with it, I’d be glad to change this part.

Relatedly, what would you suggest if we were to move the repos directory to `~/Library/Caches` (as I’ve suggested in the [original post](https://github.com/CocoaPods/CocoaPods/pull/9150#issue-315284701))? Would you provide some sort of migration for existing users or just keep existing repos where they are?

Reply:
> Relatedly, what would you suggest if we were to move the repos directory to ~/Library/Caches (as I’ve suggested in the original post)? Would you provide some sort of migration for existing users or just keep existing repos where they are?

I say it will be somewhat complicated to do now, perhaps for a 2.0 release. We _do_ use trunk now so things are even better in 1.8.x but the migration might be a bit painful. Needless to say the git repo is so huge people who have managed to clone it in bad network conditions probably don't want to try again.

> In my eyes, pod repo update doesn’t seem like the right place since this should be a one-time operation when the repos directory is created. However, if it helps users coming from older versions and you’re fine with it, I’d be glad to change this part.

I kind of agree with you but I can't find a better spot to ensure this always happens and migrates the old users as well.

Reply:
Perhaps we can add it to the `installer.rb`?

Reply:
I’ve now moved it to `pod repo update`. I think this is a good trade-off between almost never being called by existing users (`pod repo add`) and being called quite often and outside of the context of spec-repos (`pod install`).